core/templates/pages/story-editor-page/editor-tab/story-node-editor.directive.ts

```diff
@@ -116,7 +118,7 @@ angular.module('oppia').directive('storyNodeEditor', [
 
           $scope.checkCanSaveExpId = function() {
             $scope.canSaveExpId = $scope.explorationIdPattern.test(
-              $scope.explorationId);
+              $scope.explorationId) || !$scope.explorationId;
```

Comment:
Can you explain this check? (Or maybe add a comment on top)

Also, maybe rename the variable  (not start with a verb).

Reply:
The check is basically to validate an expID if a string  is added, or if it is null, mark it as saveable, as for unpublished stories 'null' expID can be saved.
For a published story, the 'delete' button is not shown, so a null expID cannot be entered by the user, in which case it would validate the expID pattern for all entered strings.

Reply:
Renamed variable.